tool/utils.ts

```diff
@@ -56,146 +57,107 @@ interface ReleaseInfo {
+ * By default, downloads the release version specified in package.json. Throws
+ * if an error occurs.
  */
-export async function getEmbeddedProtocol(options: {
-  outPath: string;
-  version?: string;
-  path?: string;
-  release?: boolean;
-}): Promise<void> {
+export async function getEmbeddedProtocol(
+  outPath: string,
+  options?:
+    | {
+        version: string;
+      }
+    | {
+        ref: string | 'main';
+      }
+    | {
+        path: string;
+      }
+): Promise<void> {
   const repo = 'embedded-protocol';
 
-  if (options.release) {
-    const latestRelease = await getLatestReleaseInfo({
-      repo,
-      tag: true,
-    });
+  if (!options || 'version' in options) {
+    const version = options?.version ?? pkg['protocol-version'];
     await downloadRelease({
       repo,
-      assetUrl:
-        `https://github.com/sass/${repo}/archive/` +
-        `${latestRelease.name.replace(' ', '-')}` +
-        ARCHIVE_EXTENSION,
+      assetUrl: `https://github.com/sass/${repo}/archive/${version}${ARCHIVE_EXTENSION}`,
       outPath: BUILD_PATH,
     });
     fs.rename(
-      p.join(BUILD_PATH, `${repo}-${latestRelease.name.replace(' ', '-')}`),
+      p.join(BUILD_PATH, `${repo}-${version}`),
       p.join(BUILD_PATH, repo)
     );
-  } else if (!options.path) {
+  }
+
+  if (options && 'ref' in options) {
     fetchRepo({
       repo,
       outPath: BUILD_PATH,
-      ref: options.version,
+      ref: options.ref,
     });
   }
 
-  const repoPath = options.path ?? p.join(BUILD_PATH, repo);
+  const repoPath =
+    options && 'path' in options ? options.path : p.join(BUILD_PATH, repo);
+
   buildEmbeddedProtocol(repoPath);
-  await linkBuiltFiles(repoPath, p.join(options.outPath, repo));
+  await linkBuiltFiles(repoPath, p.join(outPath, repo));
 }
 
 /**
- * Gets the latest version of the Dart Sass wrapper for the Embedded Compiler.
- * Throws if an error occurs.
+ * Gets the Dart Sass wrapper for the Embedded Compiler.
  *
- * @param version - If `release` is true, the version of the released binary to
- *   download (defaults to the latest version). If it's false, the Git ref to
- *   check out and build (defaults to main).
- * @param path - Build from this path instead of pulling from Github.
- * @param release - Download the latest release instead of building from source.
+ * Can download the release `version`, check out and build the source from a Git
+ * `ref`, or build from the source at `path`.
+ *
+ * By default, downloads the release version specified in package.json. Throws
+ * if an error occurs.
  */
-export async function getDartSassEmbedded(options: {
-  outPath: string;
-  version?: string;
-  path?: string;
-  release?: boolean;
-}): Promise<void> {
+export async function getDartSassEmbedded(
+  outPath: string,
+  options?:
+    | {
+        version: string;
+      }
+    | {
+        ref: string | 'main';
+      }
+    | {
+        path: string;
```

Comment:
this is not the wrong type, as the `path` option can be passed alongside others.

Thus, passing an object with extra properties is supported, so this does not prevent passing both options either. So I suggest simplifying it:

```typescript
export async function getDartSassEmbedded(
  outPath: string,
  options?: {
    version?: string;
    ref?: string;
    path?: string;
  }
)
```

Reply:
ah sorry, I misunderstood what `path` is.
But the implementation still accept reading `options.path` when it fetched the repo itself

Reply:
I understand what you're saying here. Since this is in practice called through the CLI, I'm going to prioritize unblocking that use case first, then revisit this later.